### ISSUE TITLE
New metric: histogram for image push duration

### ISSUE THREAD

**Issue by tvdfly** [superfly/flyctl#3103]
Add a new metric that records how long we spend pushing the docker image to our registry during `fly deploy`. This will help us understand regional variations and overall performance of this action and inform registry performance work.

Potential attributes:
* builder type: local, remote
* builder region: for local use the closest region (e.g. where a request hits the fly-proxy edge), for remote use the remote builder machine's region

**Comment by tvdfly** [comment]
Yep, the metrics.StartTiming() call will likely work here. We use it in a handful of places in the codebase that are good references: https://github.com/search?q=repo%3Asuperfly%2Fflyctl%20metrics.StartTiming&type=code

**Comment by utkarsh-shrivastav77** [comment]
Hi can I work on this issue
